projects/vanilla/src/app/configurators/metadata-configurator.component.ts

```diff
@@ -7,43 +7,162 @@ import { ComponentConfig, ConfiguratorContext } from "@sinequa/ngx-ui-builder";
+      <div class="mb-2">
+        <label class="form-label" for="label">Label</label>
+        <input type="text" class="form-control mb-1" id="label" [(ngModel)]="metadataToModify.label" (ngModelChangeDebounced)="configChanged()">
+      </div>
+      <div class="mb-2">
+        <sq-icon-selector [config]="metadataToModify" (configChanged)="iconChanged(metadataToModify, $event)"></sq-icon-selector>
+      </div>
+      <div class="mb-2">
+        <label class="form-label" for="fieldClass">Additional CSS classes</label>
+        <input type="text" class="form-control mb-1" id="fieldClass" [(ngModel)]="metadataToModify.fieldClass" (ngModelChangeDebounced)="configChanged()">
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="filterable" [(ngModel)]="metadataToModify.filterable" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="filterable">Filterable</label>
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="excludable" [(ngModel)]="metadataToModify.excludable" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="excludable">Excludable</label>
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="collapseRows" [(ngModel)]="metadataToModify.collapseRows" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="collapseRows">Collapse rows</label>
+      </div>
+      <div class="form-check mb-2" *ngIf="isEntity">
+        <input class="form-check-input" type="checkbox" id="showEntityExtract" [(ngModel)]="metadataToModify.showEntityExtract" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="showEntityExtract">Show entity extract</label>
+      </div>
+      <div class="mb-2" *ngIf="isEntity">
+        <label class="form-label" for="entityExtractMaxLines">Max lines for entity extract</label>
+        <input type="number" class="form-control" id="entityExtractMaxLines" [(ngModel)]="metadataToModify.entityExtractMaxLines">
+      </div>
+    </ng-container>
   </div>
 </div>
     `
 })
 export class MetadataConfiguratorComponent implements OnChanges {
   @Input() context: ConfiguratorContext;
-  @Input() metadata: MetadataConfig[];
+  @Input() metadata: string[];
 
+  modifiedMetadata: string;
+  metadataString: string;
+  metadataToModify: MetadataConfig;
+  stringToModify?: string;
+  newString: string;
   isEntity: boolean;
 
-  constructor(public appService: AppService) {}
+  constructor(public appService: AppService) { }
 
-  ngOnChanges(): void {
-    this.isEntity = this.appService.isEntity(this.config.item);
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.context && this.metadataToModify) {
+      this.metadataToModify = this.config.metadataConfig.find(m => m.field === this.metadataToModify.field);
```

Comment:
ngOnChanges is triggered when a `configChanged()` is executed and this update is needed because the object metadataToModify is referring to is overwritten otherwise so not updatable (see https://github.com/sinequa/sba-vanilla-ui-builder/pull/112#discussion_r1322969171)